commands/build.py

```diff
@@ -66,11 +66,57 @@ def make(self):
         logger.info("building & installing {}: {}".format(self.name, " ".join(cmd)))
         subprocess.check_call(cmd, env=self.env)
 
+class acbuilder(object):
+    def __init__(self, name, modules_dir, build_dir, install_dir, host_prefix, make_verbose=False):
+        self.name = name
+        self.make_verbose = make_verbose
+        self.src_path = "{}/{}".format(modules_dir, name)
+        self.build_path = "{}/{}".format(build_dir, name)
+        self.install_path = install_dir
+        self.host_prefix = host_prefix
+        self.env = os.environ.copy()
+        self.env["STAGING_DIR"] = ""
+
+    def __str__(self):
+        return "'{}' builder configuration:\n\tsrc_path: {}\n\tbuild_path: {}\n\tinstall_path: {}".format(self.name, self.src_path, self.build_path, self.install_path)
+
+    def clean(self):
+        if os.path.exists(self.src_path):
+            logger.info("cleaning {}".format(self.name))
+            os.chdir(self.src_path)
+            os.system("make clean")
+
+    def prepare(self):
+        os.chdir(self.src_path)
+        cmd = ["./build-tools/autogen.sh"]
+        logger.info("preparing {}: {}".format(self.name, " ".join(cmd)))
+        subprocess.check_call(cmd, env=self.env)
+
+        self.configure()
+
+    def configure(self):
+        cmd = ["./configure",
+               "--prefix=" + self.install_path]
+        if self.host_prefix:
+            cmd.append("--host=" + self.host_prefix)
+        logger.info("configuring {}: {}".format(self.name, " ".join(cmd)))
+        subprocess.check_call(cmd, env=self.env)
+
+    def make(self):
+        cmd = ["make", "install"]
+        if self.make_verbose:
+            cmd.extend(["V=s"])
+        else:
+            cmd.extend(["-j", str(multiprocessing.cpu_count() + 1)])
+        logger.info("building & installing {}: {}".format(self.name, " ".join(cmd)))
+        subprocess.check_call(cmd, env=self.env)
+
 class mapbuild(object):
-    def __init__(self, args):
+    @staticmethod
+    def build_map(args):
         if args.verbose: logger.setLevel(logging.DEBUG)
         commands = args.commands
-        modules = build_modules if 'all' in args.modules else [m for m in build_modules if m in args.modules]
+        modules = deps_modules if 'all' in args.deps_modules else [m for m in deps_modules if m in args.modules]
```

Comment:
Why does build_map builds deps_modules?

Reply:
yes, that's my fault. I will push the modification.